testng-core/src/test/kotlin/org/testng/internal/ClassHelperTest.kt

```diff
@@ -0,0 +1,153 @@
+import org.testng.internal.samples.classhelper.issue1339.LittlePandaSample
+import org.testng.internal.samples.classhelper.issue1456.TestClassSample
+import org.testng.internal.samples.classhelper.misamples.AbstractMovesSample
+import org.testng.internal.samples.classhelper.misamples.BatmanSample
+import org.testng.internal.samples.classhelper.misamples.JohnTravoltaMovesSample
+import org.testng.internal.samples.classhelper.misamples.MickJaggerSample
+import org.testng.xml.XmlClass
+import org.testng.xml.XmlSuite
+import org.testng.xml.XmlTest
+import kotlin.reflect.KClass
+
+class ClassHelperTest {
+
+    @Test(dataProvider = "testDataProvider")
+    fun runFindClassesInSameTestScenarios(
+        scenarioName: String,//This parameter will help us differentiate scenarios in reports
+        classToBeFound: KClass<*>,
+        expectedCount: Int, vararg classes: KClass<*>
+    ) {
+        Reporter.log("Running Scenario $scenarioName")
+        val xmlSuite = XmlSuite().apply {
+            name = "xml_suite"
+        }
+        listOf("test1", "test2", "test3").forEach {
+            xmlSuite.addXmlTest(it, *classes)
+        }
+        val xmlClasses = ClassHelper.findClassesInSameTest(classToBeFound.java, xmlSuite)
+        assertThat(xmlClasses).hasSize(expectedCount)
+    }
+
+    @DataProvider(name = "testDataProvider")
+    fun testDataProvider(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf("FindClassInSameTest", TestClassSample::class, 1, TestClassSample::class),
+            arrayOf(
+                "FindClassesInSameTest",
+                TestClassSample::class, 2, TestClassSample::class, BabyPandaSample::class
+            ),
+        )
+    }
+
+    @Test(dataProvider = "scenariosData")
+    fun runAvailableMethodsScenarios(
+        scenarioName: String,//This parameter will help us differentiate scenarios in reports
+        expected: List<String>,
+        whichClass: KClass<*>
+    ) {
+        Reporter.log("Running Scenario $scenarioName")
+        val methods = ClassHelper.getAvailableMethods(whichClass.java)
+            .filter {
+                //jacocoInit() is something that Gradle adds. So filter it out.
+                !"\$jacocoInit".contentEquals(it.name)
+            }.toList()
+        val size = expected.size
+        assertThat(methods.size)
+            .withFailMessage("Number of methods found should have been at-least $size")
+            .isEqualTo(size)
+        val actual = methods.map {
+            it.name
+        }.toList()
+        assertThat(actual).containsAll(expected)
+    }
+
+    @DataProvider(name = "scenariosData")
+    fun scenariosData(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf("AllAvailableMethods", getExpected(), LittlePandaSample::class),
+            arrayOf(
+                "AvailableMethodsWhenOverridingIsInvolved",
+                getExpected("equals", "hashCode", "toString"),
```

Comment:
`getExpected() + listOf("equals", "hashCode", "toString")` looks more readable because there is no possible side-effects on the provided list.

Reply:
fixed